rust/lance/src/io/exec/planner.rs

```diff
@@ -295,8 +296,8 @@ impl Planner {
     fn parse_sql_expr(&self, expr: &SQLExpr) -> Result<Expr> {
         match expr {
             SQLExpr::Identifier(id) => {
-                if id.quote_style == Some('"') {
-                    Ok(Expr::Literal(ScalarValue::Utf8(Some(id.value.clone()))))
+                if id.quote_style == Some('`') {
+                    Ok(Expr::Column(Column::from_name(id.value.clone())))
```

Comment:
I think the first part we want to keep: it's there to parse double quotes into string literals.
Let's add some comments here to avoid future confusion 😀 
```suggestion
                // Users can pass string literals wrapped in `"`.
                // (Normally SQL only allows single quotes.)
                if id.quote_style == Some('"') {
                    Ok(Expr::Literal(ScalarValue::Utf8(Some(id.value.clone()))))
                // Users can wrap identifiers with ` to reference non-standard
                // names, such as uppercase or spaces.
                } else if id.quote_style == Some('`') {
                    Ok(Expr::Column(Column::from_name(id.value.clone())))
```